### PR TITLE
[Detox] update test config

### PR DESCRIPTION
**Pull request by cortisiko**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

In order to have the detox tests running on jest v27, we need to migrate to detox 20. This PR accomplishes that by following the migration guide [here](https://wix.github.io/Detox/docs/guide/migration/#jest-2725-and-higher). 

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: e2e/init.js [hunk 1]
```diff
@@ -1,22 +1,9 @@
...
-
 beforeAll(async () => {
-  await detox.init(config);
   await device.launchApp();
 });
 
-beforeEach(async () => {
-  await adapter.beforeEach();
-});
+beforeEach(async () => {});
```
Comment (Cal-L) [inline e2e/init.js:5]: Can remove whole block

**Review by Cal-L — APPROVED** [review]
Just left a small nit. LGTM

File: e2e/init.js (hunk 1 above)
Comment (cortisiko) [reply e2e/init.js]: okay, updated!

**Comment by cortisiko** [comment]
@SocketSecurity ignore-all

**Pull request merged**